evalml/pipelines/components/transformers/samplers/undersampler.py

```diff
@@ -0,0 +1,93 @@
+import pandas as pd
+from skopt.space import Integer, Real
+
+from evalml.pipelines.components.transformers import Transformer
+from evalml.preprocessing.data_splitters.balanced_classification_sampler import (
+    BalancedClassificationSampler
+)
+from evalml.utils.woodwork_utils import (
+    _convert_woodwork_types_wrapper,
+    infer_feature_types
+)
+
+
+class Undersampler(Transformer):
+    """Random undersampler component."""
+    name = "Undersampler"
+    hyperparameter_ranges = {
+        'balanced_ratio': Real(1, 10),
+        'min_samples': Integer(50, 1000),
```

Comment:
I didn't put `min_percentage` since that's only used in conjunction with determining if the class is severely imbalanced, in which case we don't sample. `min_samples` controls both the severely imbalanced case as well as the imbalanced case by limiting the amount we can downsample a certain class to. I figured `min_samples` would be more useful to tune than `min_percentage` for this reason. 
Becausse of this, I'm not sure how useful it would be to expose `min_percentage` as a hyperparam for `AutoMLSearch` to iterate though. What do you think @ParthivNaresh?